client/python/openlineage/client/transport/msk_iam.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import functools
+import logging
+import os
+from typing import Any
+
+import attr
+from openlineage.client.transport import KafkaConfig, KafkaTransport
+
+log = logging.getLogger(__name__)
+
+
+def _detect_running_region() -> None | str:
+    """Dynamically determine the region from a running Glue job (or anything on EC2 for
+    that matter).
+    https://stackoverflow.com/questions/37514810/how-to-get-the-region-of-the-current-user-from-boto
+    """
+    import boto3  # type: ignore[import]
+
+    easy_checks = [
+        # check if set through ENV vars
+        os.environ.get("AWS_REGION"),
+        os.environ.get("AWS_DEFAULT_REGION"),
+        # else check if set in config or in boto already
+        boto3.DEFAULT_SESSION.region_name if boto3.DEFAULT_SESSION else None,
+        boto3.Session().region_name,
+    ]
+    region: None | str = None
+    for region in easy_checks:
+        if region:
+            return region
+
+    # else query an external service
+    # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+    import requests
+
+    try:
+        r = requests.get("http://169.254.169.254/latest/dynamic/instance-identity/document", timeout=1)
```

Comment:
Yeah, I can add a link to the documentation of the [instance metadata](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-dynamic-data-retrieval.html) to explain why there is a fixed IP in the code.

In short, it's an HTTP server running on EC2 where you can get metadata information on the instance itself.
Now that you point it out, I should also support `IMDSv2` 

So problably It's better to drop it because it's just a nice to have but not bring much value.